modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaHttpServerCodegen.java

```diff
@@ -0,0 +1,486 @@
+package org.openapitools.codegen.languages;
+
+import io.swagger.v3.oas.models.Operation;
+import io.swagger.v3.oas.models.parameters.Parameter;
+import io.swagger.v3.oas.models.servers.Server;
+import org.openapitools.codegen.*;
+
+import java.io.File;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.openapitools.codegen.meta.features.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ScalaAkkaHttpServerCodegen extends AbstractScalaCodegen implements CodegenConfig {
+    protected String groupId;
+    protected String artifactId;
+    protected String artifactVersion;
+    protected String invokerPackage;
+
+    protected String akkaHttpVersion;
+
+    public static final String AKKA_HTTP_VERSION = "akkaHttpVersion";
+    public static final String AKKA_HTTP_VERSION_DESC = "The version of akka-http";
+    public static final String DEFAULT_AKKA_HTTP_VERSION = "10.1.10";
+
+    static Logger LOGGER = LoggerFactory.getLogger(ScalaAkkaHttpServerCodegen.class);
+
+    public CodegenType getTag() {
+        return CodegenType.SERVER;
+    }
+
+    public String getName() {
+        return "scala-akka-http";
```

Comment:
I'll rename it to "scala-akka-http-server" to follow our generator naming convention.